framework/src/problems/FEProblemBase.C

```diff
@@ -2284,7 +2284,7 @@ FEProblemBase::projectSolution()
   _aux->solution().localize(*_aux->sys().current_local_solution, _aux->dofMap().get_send_list());
 }
 
-std::shared_ptr<Material>
+std::shared_ptr<MaterialBase>
 FEProblemBase::getMaterial(std::string name,
                            Moose::MaterialDataType type,
                            THREAD_ID tid,
```

Comment:
Why would not we have `getMaterial`  for the normal `Material`s and `getInterfaceMaterial` for the new one? That would avoid breaking the API, so the user code could stay unchanged. 

Another issue. It does not seem that `MaterialDataType` got a new value for the interface materials. So how would one get Interface materials via this call?  It may not be need though to add a new value. If we go with 2 methods: one for volumetric/side/neighbor that returns `Material` and the other that returns `InterfaceMaterials`. But you would still have to do some work in `MaterialWarehouse`, to get the right list back and also store interface materials in their own container

Reply:
Looking at the code changes this triggered, definitely have 2 methods.